README.md

```diff
@@ -1,282 +1,257 @@
-    console.log(e)
-});
-```
-
-#### Get all shared accounts
-**`SharedAccount` is an object representing the `public address`, `type`, `isMyAccount`, `nickname`and `userId`** -- All the accounts you share to others and the accounts shared to you
-```
-var sharedAccounts = vault.getSharedAccounts()
+val walletAccount = vault.getWalletAccount(keyAlias)
 ```
-#### Observing shared accounts
+> * keyAlias - 
+#### Observing **my** Wallet Key Accounts
+To setup a listenter for one of your wallets, so that you can recieve notifcations of changes in wallet activity, use the following call. 
 ```
-vault.startObservingSharedAccounts(callback)
-...
-vault.stopObservingSharedAccounts(callback)
+DeviceEventEmitter.addListener(walletAccounts)
 ```
+> * walletAccounts - 
 #### Delete/Stop tracking the account
+Delete a key account from the current user. This deletes the private key locally as well as all meta data associated with this key account. 
 ```
 vault.deleteAccount(walletAccount)
 ```
-### Mnemonic
-#### Generate a seed (word size can be 16, 20, 24, 28 or 32)
-```
-var generatedMnemonicSeed = vault.generateSeed(wordSize)
-```
-#### Import from seed (or generate). 
-*This will register the `WalletAccount` automatically with the backend. The key alias can be public -- no need to treat it like a password.*
-```
-var keyAlias = vault.importMnemonic(menmonicString, addressIndex = 0)
-```
-### Get WalletAccount for KeyAlias
-```
-val walletAccount = vault.getWalletAccount(keyAlias)
-```
-#### Get Public Address of a Mnemonic
-*CryptoObject is used in conjunction with the SE/TEE to validate user biometric authentication (optional).*
-
-
-*We are decomposing the key alias which has the public address of address 0 of the mnemonic*
+> * walletAccount -
+### Shared Accounts
+#### Share a key account with user
+Expose a current account as a shared account to a specified user on the service. 
 ```
-val publicAddress = vault.getPublicAddress(mnemonicString, addressIndex = 0)
-// OR
-val publicAddress = vault.getPublicAddress(keyAlias, addressIndex = 0, cryptoObject?)
-```
-
-#### Generate Another wallet from a mnemonic
-*Automatically add the account address to backend*
+vault.shareAccount(userId, walletAccount)
 ```
-var publicAddress = vault.createWallet(keyAlias, nameOfWallet, addressIndex = 0, cryptoObject?)
-// OR
-var publicAddress = vault.createWallet(mnemonicString, nameOfWallet, addressIndex = 0, cryptoObject?)
+> * userId - 
+> * walletAccount - 
+#### Unshare an key account with user
+Remove visibility of a shared account from a specified user on the service. 
 ```
-
-### Get Transactions
+vault.unshareAccount(userId, walletAccount)
+````
+> * userId - 
+> * walletAccount - 
+#### Get all shared key accounts relationships
+Retrieve a list of all of the currently shared key accounts assoicated with the current user? 
+@Raul Some details on the context of this call? Does it take param for a user, is there some user state? 
```

Comment:
there shouldn't be any specific parameters here, this will just return all the the shared accounts. imagine i have 10 friends and i have shared different wallets for different friends

Reply:
something like userAccount.getAll...